src/features/offer/components/OfferContent/OfferContent.tsx

```diff
@@ -1,203 +1,57 @@
-          isVisible={visible}
-          offerImages={offerImages}
-          defaultIndex={carouselDefaultIndex}
-        />
-      ) : null}
-
-      <ScrollViewContainer
-        testID="offerv2-container"
-        scrollEventThrottle={16}
-        scrollIndicatorInsets={scrollIndicatorInsets}
-        bounces={false}
-        onScroll={onScroll}>
-        {isDesktopViewport ? (
-          <BodyDesktopContainer headerHeight={headerHeight} testID="offer-body-desktop">
-            <OfferImageContainer
-              imageUrls={offerImages}
-              categoryId={subcategory.categoryId}
-              shouldDisplayOfferPreview={enableOfferPreview}
-              onPress={onPress}
-            />
-            <OfferBody
-              offer={offer}
-              subcategory={subcategory}
-              trackEventHasSeenOfferOnce={trackEventHasSeenOfferOnce}
-            />
-          </BodyDesktopContainer>
-        ) : (
-          <ViewGap gap={8} testID="offer-body-mobile">
-            <OfferImageContainer
-              imageUrls={offerImages}
-              categoryId={subcategory.categoryId}
-              shouldDisplayOfferPreview={enableOfferPreview}
-              onPress={onPress}
-            />
-            <OfferBody
-              offer={offer}
-              subcategory={subcategory}
-              trackEventHasSeenOfferOnce={trackEventHasSeenOfferOnce}
-            />
-          </ViewGap>
-        )}
+  const footer = (
+    <OfferCTAButton
+      offer={offer}
+      subcategory={subcategory}
+      trackEventHasSeenOfferOnce={trackEventHasSeenOfferOnce}
+    />
+  )
 
-        <OfferPlaylistList
-          offer={offer}
-          position={userLocation}
-          sameCategorySimilarOffers={sameCategorySimilarOffers}
-          apiRecoParamsSameCategory={apiRecoParamsSameCategory}
-          otherCategoriesSimilarOffers={otherCategoriesSimilarOffers}
-          apiRecoParamsOtherCategories={apiRecoParamsOtherCategories}
-          sameArtistPlaylist={sameArtistPlaylist}
-        />
-        {isDesktopViewport ? null : <Spacer.Column numberOfSpaces={22} />}
-      </ScrollViewContainer>
-      {/* OfferHeader is called after Body to implement the BlurView for iOS */}
-      {isWeb ? null : (
-        <OfferHeader title={offer.name} headerTransition={headerTransition} offer={offer} />
-      )}
-      {isDesktopViewport ? null : (
-        <OfferCTAButton
-          offer={offer}
-          subcategory={subcategory}
-          trackEventHasSeenOfferOnce={trackEventHasSeenOfferOnce}
-        />
+  return (
+    <OfferContentBase
+      offer={offer}
+      searchGroupList={searchGroupList}
+      showOfferPreview={showOfferPreview}
+      contentContainerStyle={CONTENT_CONTAINER_STYLE}
+      onOfferPreviewPress={handlePress}
+      footer={footer}
+      subcategory={subcategory}>
+      {(body) => (
+        <ViewGap gap={8} testID="offer-body-mobile">
+          {body}
+        </ViewGap>
       )}
```

Comment:
avec mon précédent commentaire

ça donnerait

```suggestion
      {TrouveUnNom}
```

avec

```tsx
const TrouveUnNom = styled(ViewGap).attrs({
        gap: 8,
        testID: "offer-body-mobile",
      })``
```